engine/src/cubos/engine/plugins/window.cpp

```diff
@@ -0,0 +1,42 @@
+#include <cubos/engine/plugins/window.hpp>
+#include <cubos/core/settings.hpp>
+
+static void startup(cubos::core::io::Window& window, ShouldQuit& quit, cubos::core::Settings& settings)
+{
+    quit.value = false;
+    window = cubos::core::io::openWindow(
+        settings.getString("window.title", "Cubos"),
+        {
+            settings.getInteger("window.width", 800),
+            settings.getInteger("window.height", 600)
+        }
+    );
+}
+
+static void pollSystem(const cubos::core::io::Window& window, ShouldQuit& quit)
+{
+    window->pollEvents();
```

Comment:
Possible mistake here: I don't know if the inner GLFW implementation is thread safe.
Can `pollEvents` really be called at the same time as `swapBuffers`? We should check that, and if not, we should make the methods require non-const access to the Window, which should prevent users from doing unsafe stuff like this.

Reply:
Supposedly swap buffer only happens after polling and after all user code and engine rendering happens.

Reply:
Okay we need to take that into account when adding multi threading to the dispatcher.
Maybe we should add a way to force a system to run on a specific thread?

Reply:
#241

Reply:
> Okay we need to take that into account when adding multi threading to the dispatcher. Maybe we should add a way to force a system to run on a specific thread?

Seems like we have no option. [This post](https://discourse.glfw.org/t/multithreading-glfw/573) mentions how "the OS only posts events to the thread where the window was created so polling needs to happen on that thread". So it's another config required for the dispatcher.